scipy/signal/_spectral_py.py

```diff
@@ -153,8 +153,9 @@ def lombscargle(x,
     return pgram
 
 
-def periodogram(x, fs=1.0, window='boxcar', nfft=None, detrend='constant',
-                return_onesided=True, scaling='density', axis=-1):
+def periodogram(x, fs=1.0, window='boxcar', nperseg=None, nfft=None,
```

Comment:
As a general principle, you can only ever add arguments to the end of function signatures for public functions (unless they're keyword-only). Otherwise you are breaking backwards compatibility.